### PR TITLE
Pya 38 events page

### PR DESCRIPTION
**Pull request by cristoph143**
# Pull Request Description

## Summary
This pull request includes several updates and enhancements to the event page template and related files. The primary focus is on improving the layout, functionality, and visual presentation of the event section. Key changes include refactoring CSS and HTML templates, adding new JSON data files for events and sponsors, and enhancing the user experience with better navigation and image transitions.

## Changes Made
1. **Event Page Template:**
   - Refactored to include a new hero section for better visual presentation.
   - Added fixed and resize CSS classes for improved layout control.

2. **Functionality Enhancements:**
   - Added a smooth loading transition effect for event images.
   - Implemented functionality to open event links in new tabs.
   - Verified that all event links consistently open in new tabs.

3. **Content and Data Updates:**
   - Added and updated JSON data files for events, history, and sponsors.
   - Reduced the number of event cards displayed to three and updated the links to ensure they point to the correct destinations.
   - Added new sponsor and event images under the `/static/images/` directory.

4. **Code Merges:**
   - Merged the latest changes from the 'develop' branch into the PYA-38-Events-Page branch to synchronize updates.

## Testing
- Verified that the new hero section is correctly displayed on the event page.
- Confirmed that the new CSS classes improve layout responsiveness and fixed element positioning.
- Tested the loading transition effect for event images to ensure smooth visual transitions.
- Checked that all event links open in new tabs and navigate to the correct destinations.
- Ensured that the event cards display only three items and link to the updated URLs.
- Confirmed that new sponsor and event images are correctly loaded and displayed.
- Reviewed merged changes from the 'develop' branch to ensure no conflicts or issues.

### CODE REVIEW

File: PizzaPyWebApp/templates/partials/events/hero.html [hunk 1]
```diff
@@ -0,0 +1,128 @@
+<div class="relative flex flex-col items-center justify-center w-full">
+    <img src="../../static/images/python-3d.png" alt="PizzaPy Logo"
+        class="absolute scale-[1.5] md:scale-[2.0] lg:scale-[3.5] rotate-[11deg] opacity-30 blur-xl -z-50 lg:right-[20%] lg:bottom-[50%]" />
+    <div class="flex flex-col items-center w-full my-10 lg:my-20">
+        <div class="font-bold text-white text-[36px] md:text-[48px] lg:text-[72px] text-center">What’s happening now?
```
Comment (Luzefiru) [inline PizzaPyWebApp/templates/partials/events/hero.html:5]: Use proper semantic HTML, in this case, `<h1>` tag to denote the main title of the page.

File: PizzaPyWebApp/templates/partials/events/hero.html [hunk 2]
```diff
@@ -0,0 +1,128 @@
+<div class="relative flex flex-col items-center justify-center w-full">
+    <img src="../../static/images/python-3d.png" alt="PizzaPy Logo"
+        class="absolute scale-[1.5] md:scale-[2.0] lg:scale-[3.5] rotate-[11deg] opacity-30 blur-xl -z-50 lg:right-[20%] lg:bottom-[50%]" />
+    <div class="flex flex-col items-center w-full my-10 lg:my-20">
```
Comment (Luzefiru) [inline PizzaPyWebApp/templates/partials/events/hero.html:4]: Looks too tight on mobile devices. Perhaps add some padding?

On that note, line height is too spaced out. Would making it tighter improve it?

image

File: PizzaPyWebApp/templates/partials/events/hero.html (hunk 1 above)
Comment (Luzefiru) [inline PizzaPyWebApp/templates/partials/events/hero.html:5]: Use proper semantic HTML, in this case, `<h1>` tag to denote the main title of the page.

**Review by Luzefiru — CHANGES_REQUESTED** [review]
Left a few comments on spacing after testing the application on various screen devices. Amazing work so far and keep up the good work!

File: PizzaPyWebApp/templates/partials/events/hero.html (hunk 2 above)
Comment (cristoph143) [reply PizzaPyWebApp/templates/partials/events/hero.html:4]: https://github.com/pizzapy/pizzapy-website/assets/42131558/2bb8025f-0eac-4570-aae2-69312d5c084d

**Pull request merged**